contrib_bots/lib/hangman.py

```diff
@@ -0,0 +1,167 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+import requests
+from six.moves import range
+
+
+class HangmanHandler(object):
+    def usage(self):
+        return '''
+    This bot allows users to play Hangman in streams that the Hangman
+    bot is subscribed to using the following simple commands:
+    - @hangman new: This command starts a new Hangman game.
+    - @hangman guess <letter>: This command lets the user
+    guess a letter for the current Hangman game.
+    - @hangman stop: This command stops the current Hangman game.
+    - @hangman help: This command gives users information
+    about the Hangman bot.
+    '''
+
+    def triage_message(self, message, client):
+        # Return True if we want to (possibly) respond to this message
+        original_content = message['content']
+        is_hangman_command = original_content.lower().startswith('@hangman ')
+
+        return is_hangman_command
+
+    def get_new_word(bot_self):
+        link = "http://www.setgetgo.com/randomword/get.php"
+        try:
+            bot_self.currentword
+        except AttributeError:
```

Comment:
I think `if hasattr(bot_self, 'current_word'):` will be more appropriate here. Exception handling is more expensive resources wise and should be reserved for 'exceptional situations' :).